### PR TITLE
New dbconfig parameter replaces all per db options.

### PR DESCRIPTION
**Pull request by traylenator**
#### Pull Request (PR) description

Previously each database was configured via a flat
name space of variables e.g.

* passwd_positive_ttl positive time to live for passwords database.
* passwd_negative_ttl negative time to live for passwords database.
* group_positive_ttl positive time to live for groups database.
* group_negative_ttl negative time to live for groups database.
* hosts_positive_ttl positive time to live for hosts database.
* hosts_negative_ttl negative time to live for hosts database.

These are now all moved into a hash of configuration.

```
class{'nscd':
  dbconfig => {
    'passwd' => {
      'enable-cache' => true,
      'positive-time-to-live' => 500,
      'negative-time-to-live' => 500,
     },
    'group' => {
      'enable-cache' => true,
      'positive-time-to-live' => 500,
      'negative-time-to-live' => 500,
     },
  }
  },
}
```

In addition new parameters have been added to specify the running user
which defaults correctly on Debian based machines.

#### This Pull Request (PR) fixes the following issues
Fixes: #40 
Fixes: #15 
Fixes: #31 
Fixes: #30

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -3,47 +3,48 @@
...
-  Integer                           $hosts_negative_ttl    = 20,
-  Integer                           $hosts_positive_ttl    = 3600,
-  Integer                           $services_negative_ttl = 20,
-  Integer                           $services_positive_ttl = 28800,
+  Nscd::Database                    $dbconfig,
+  Enum['present','absent','latest'] $pkg_ensure       = 'present',
+  Boolean                           $service_ensure   = true,
+  Boolean                           $service_enable   = true,
+  String                            $user             = 'root',
+  Optional[String]                  $stat_user        = undef,
```
Comment (bastelfreak) [inline manifests/init.pp]:
```suggestion
  Optional[String[1]]                  $stat_user     = undef,
```

File: metadata.json [hunk 2]
```diff
@@ -27,18 +27,26 @@
     }
   ],
   "operatingsystem_support": [
+    {
+      "operatingsystem": "Ubuntu",
+      "operatingsystemrelease": [
+        "16.04"
```
Comment (bastelfreak) [inline metadata.json]: did you check if 18.04 works as well?

File: spec/acceptance/class_spec.rb [hunk 3]
```diff
@@ -13,6 +13,97 @@ class { 'nscd':}
       apply_manifest(pp, catch_changes: true)
     end
 
+    describe package('nscd') do
```
Comment (bastelfreak) [inline spec/acceptance/class_spec.rb]: more acceptance tests \o/

File: metadata.json (hunk 2 above)
Comment (traylenator) [reply metadata.json]: debian 8 added also. ( though testing it locally destroyed my laptop :-( )

**Comment by traylenator** [comment]
On CentOS we are ending up with 

restart-interval   

with no value .... Tests are clearly deficient.

**Comment by traylenator** [comment]
restart-interval problem fixed now.

**Comment by bastelfreak** [comment]
Thanks for the work!

**Pull request merged**